crates/swift-bridge-ir/src/codegen/generate_rust_tokens/shared_enum.rs

```diff
@@ -130,14 +130,31 @@ impl SwiftBridgeModule {
             convert_ffi_variants_to_rust.push(convert_ffi_variant_to_rust);
         }
 
-        // TODO:
-        //  Parse any derives that the user has specified and combine those with our auto derives.
-        let automatic_derives = if shared_enum.has_one_or_more_variants_with_data() {
+        // Auto derives
+        let mut derives = if shared_enum.has_one_or_more_variants_with_data() {
             vec![]
         } else {
             vec![quote! {Copy}, quote! {Clone}]
         };
 
+        // User derives
+        let mut derive_impls = vec![];
+
+        if shared_enum.derive.debug {
```

Comment:
Can we, for now, add a `&& !shared_enum.has_one_or_more_variants_with_data()`.

Reason being that our current `Debug` implementation might lead people to making memory safety mistakes when using enums that hold non `Copy` data .

For example, say you have the following enum.

```rust
#[swift_bridge::bridge]
mod ffi {
    enum MyEnum {
        Variant(String)
    }
}
```

Right now we don't have a way to pass enums by reference. 
So, right now passing an enum from `Swift` -> `Rust` means passing ownership from `Swift` -> `Rust`.

So, if you were to call the enum's Debug implementation you would be passing ownership of the enum, and thus the enum's inner data, to Rust.

So, if you tried to use the inner data after calling the `Debug` implementation you would trigger undefined behavior.

---

At some point we'll want to design a way to pass enums between `Rust`/`Swift` by reference.. 
But for the purpose of this PR we can just limit `Debug` to enums that implement `Copy/Clone`.

Reply:
[`e1a701a` (#194)](https://github.com/chinedufn/swift-bridge/pull/194/commits/e1a701a4973faef13a593d26d8270954edfbf598)